demo/src/components/CorefComponent.js

```diff
@@ -0,0 +1,203 @@
+            </div>
+            <div className="form__field form__field--btn">
+            <Button outputState={outputState} runModel={runCorefModel} inputs={corefInputs} />
+            </div>
+        </div>
+        );
+    }
+}
+
+/*******************************************************************************
+  <CorefOutput /> Component
+*******************************************************************************/
+
+class CorefOutput extends React.Component {
+    render() {
+      const { document, clusters } = this.props;
+      return (
+        <div className="model__content">
+          <div className="form__field">
+            <label>Clusters</label>
+            <div className="model__content__summary">
+            <ul>
+              {this.props.clusters.map((cluster) =>
+               <li>
+                {cluster.map((span) =>
+                <span> {this.props.document.slice(span[0], span[1] + 1).join(" ")}, </span>
+                )}
+               </li>
+            )}
+            </ul>
+            </div>
+          </div>
+
+          <div className="form__field">
+            <label>Document</label>
+            <div className="passage model__content__summary">
+            {this.props.document.map((word, index) =>
+              <span> {word} </span>
+            )}
+            </div>
+          </div>
+        </div>
+      );
+    }
+  }
+
+
+/*******************************************************************************
+  <CorefComponent /> Component
+*******************************************************************************/
+
+class CorefComponent extends React.Component {
+    constructor() {
+      super();
+
+      this.state = {
+        outputState: "empty", // valid values: "working", "empty", "received", "error"
+        document: "",
+        clusters: {},
+      };
+
+      this.runCorefModel = this.runCorefModel.bind(this);
+    }
+
+    runCorefModel(event, inputs) {
+      this.setState({
+        outputState: "working",
+      });
+
+      var payload = {
+        document: inputs.documentValue,
+      };
+
+      fetch('/predict/coreference-resolution', {
+        method: 'POST',
+        headers: {
+          'Accept': 'application/json',
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(payload)
+      }).then((response) => {
+        return response.json();
+      }).then((json) => {
+        this.setState({clusters: json["clusters"]});
+        this.setState({document: json["document"]});
+        this.setState({outputState: "received"});
+      }).catch((error) => {
+        this.setState({outputState: "error"});
+        throw error; // todo(michaels): is this right?
```

Comment:
@codeviking someone had this question at some point, and now it's been copy-pasted through our 4 examples.  Is this a reasonable thing to do, or should we somehow log our JavaScript exception without throwing it?

Reply:
In this case I'd replace it with `console.error(error)`.  The thrown exception can't actually be caught, as the exception is thrown within an asynchronous block.   In other words, this wouldn't work.

```
try {
  runCorefModel(...)
} catch(err) {
  // You won't catch it here.
}
```

If you wanted the invoker to be able to handle the exception I'd simply return the result of `fetch` (so that the implementor can make a decision around control flow).  That said it doesn't seem like you need that flexibility here, so I'd just dump it to the console and call it good :).

Reply:
What are the negative consequences of uncaught Javascript exception @codeviking ?

Reply:
On the client, none.  It'll just propagate to a `console.log` ultimately.

On the server it'll cause the program to crash, which isn't great if it's an HTTP server :).